### ISSUE TITLE
scheduler.step() is called before optimizer.step()

### ISSUE THREAD

**Issue by SamSandwich07** [lincc-frameworks/fibad#983]
**Bug report**
Scheduler is called before optimizer:
```python
/hyrax/src/hyrax/pytorch_ignite.py:756: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`.
In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`. 
Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.
See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
    model.scheduler.step()
```
Reproduce by running the following test: `test_train.py::test_training_info_returned_on_model`. Maybe other tests will cause this issue as well, I'd be surprised if they don't.

Should be reproducible on any environment.

</details>

**Before submitting**
Please check the following:

- [x] I have described the situation in which the bug arose, including what code was executed, and any applicable data others will need to reproduce the problem.
- [x] I have included information about my environment, including the version of this package (e.g. `hyrax.__version__`)
- [x] I have included available evidence of the unexpected behavior (including error messages, screenshots, and/or plots) as well as a description of what I expected instead.
- [ ] If I have a solution in mind, I have provided an explanation and/or pseudocode and/or task list.

**Comment by SamSandwich07** [comment]
After some further inspection, I am thinking this might be an issue particular to Hyrax Loopback because it doesn't define an optimizer. I'll do some more investigation on this once the #996 is merged because the test changes slightly there and I don't want to mess with an old version.

**Comment by SamSandwich07** [comment]
I have concluded that this is actually just an artifact of the `HyraxLoopback` model not having an optimizer. In any practical model, there will be one included and it should be called before the scheduler (since `optimizer.step()` should be in `train_batch` and `scheduler.step()` is called after that).
